Objects/unionobject.c

```diff
@@ -418,14 +436,47 @@ union_repr(PyObject *self)
     return NULL;
 }
 
+static PyObject *
+union_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    unionobject *alias = (unionobject *)self;
+    PyObject* from_args = PyObject_GetAttrString(self, "_from_args");
+    if (from_args == NULL) {
+        return NULL;
+    }
+
+    return Py_BuildValue("O(O)", from_args, alias->args);
```

Comment:
@pablogsal and @uriyyo the reference leak is here, we need to decref ``from_args``. I am submitting a PR now.

Reply:
Yeah, I just catched it as well: https://github.com/python/cpython/pull/27332

Reply:
Thanks @Fidget-Spinner for looking at it!

Reply:
@Fidget-Spinner thanks for solving this issue